drud/ddev
# Allow config.*.y*ml files to override config.yaml, fixes #1410

## The Problem/Issue/Bug:

OP #1410 requests config.yaml overrides. There are lots of different opinions in the thread, so a very simple approach is used here.

## How this PR Solves The Problem:

* .ddev/config.yaml is loaded first
* After that, override files in .ddev/config.*.y*ml are loaded, in alphabetical/glob order. 

Users can choose the workflow they want for these and can choose how to manage them in git.

The simplest possible technique could be to use a `config.local.yaml` for things that should not be shared with the team.

## TODO:

- [x] Add `ddev debug configyaml` to show the resultant config.yaml totally loaded, and also make sure it shows what files are being loaded.
- [x] Make sure that `ddev config` and anything else that writes config.yaml *only* writes stuff that was in the config.yaml, not in the add-ons. Probably need a test for this.
- [x] Add docs on how this works

## Manual Testing Instructions:

* Add a `.ddev/config.local.yaml` with custom configuration. `ddev start` and inspect the results.
* Try this with multiple files.
* Test multiple variants of `ddev config` (which writes out config.yaml) to make sure that config.*.yaml values cannot sneak into the written config.yaml.
* Verify the correctness of the output of `ddev debug configyaml`
* Verify the correctness of new docs

## Automated Testing Overview:

- [x] Test this basic behavior
- [x] Test that config.*.yaml are properly ordered by the glob on all test environments.

## Related Issue Link(s):

OP #1410

## Release/Deployment notes:
